Kubernetes/windows/debug/collectlogs.ps1

```diff
@@ -38,14 +39,49 @@ md $outDir
 pushd 
 cd $outDir
 
-Get-HnsNetwork | Select Name, Type, Id, AddressPrefix > network.txt
-Get-hnsnetwork | Convertto-json -Depth 20 >> network.txt
-Get-hnsnetwork | % { Get-HnsNetwork -Id $_.ID -Detailed } | Convertto-json -Depth 20 >> networkdetailed.txt
+# V1
+if($HnsSchemaVersion -eq 1)
+{
+    Get-HnsNetwork -Version 1 | Select Name, Type, Id, @{Name="AddressPrefix"; Expression={$_.Subnets.AddressPrefix}} > network.txt
+    Get-HnsEndpoint -Version 1 | Select IpAddress, MacAddress, IsRemoteEndpoint, State > endpoint.txt
+    Get-HnsNamespace -Version 1 `
+    | Select ID, `
+    CompartmentId, `
+    CompartmentGuid, `
+    IsDefault, @{Name="EndpointID"; Expression={$_.ResourceList.Data.Id}} > namespaces.txt
+}
+# V2
+else
+{
+    # V2 Networks
+    Get-HnsNetwork -Version 2 `
+    | Select Name, Type, Id,`
+    @{Name="AddressPrefix"; Expression={$_.Ipams.Subnets.IpAddressPrefix}}` > network.txt
+
+    # V2 Endpoints
+    Get-HnsEndpoint -Version 2 `
+    | Select @{Name="IpAddress"; Expression={$_.IpConfigurations.IpAddress}}, `
+    MacAddress, `
+    (@{Name="IsLocal"; Expression={$_.Health.Extra.Resources.Allocators.IsLocal | Where-Object {$_ -ne $null}}}), `
```

Comment:
This field does not get populated if v1 HNS endpoints were created with Azure CNI on Docker. Wondering if we should just use the V1 equivalent (negated) field "IsRemoteEndpoint" in all cases until official v1.23 deprecation.

Reply:
I will be using the v1 IsRemoteEndpoint instead.